### DISCUSSION TITLE
Where should I add `lib_deps`

### DISCUSSION THREAD

**Discussion by bilogic** [arendst/Tasmota#12020] [Q&A] [answered]
Hi,

I'm trying to add an external library using `lib_deps`. In simpler projects, I add it as follows to my `platformio.ini`
```
[env:esp12e]
...
lib_deps =
    1002                     ;elapsedMillis
    Servo(esp8266)    ;the native library provided by Arduino
```

I tried `platformio_override.ini` under `[common]` and `[core]`, but had no effect, basically throws a fatal error at the line where I tried to include the header files. Which file should I add it to in Tasmota? 

Thank you.

**Comment by Jason2866** [comment, marked as answer]
Servo lib is deactivated in Tasmota PlatformIO setup. https://github.com/arendst/Tasmota/blob/development/platformio_tasmota_env.ini#L19-L35 
Not needed and not intended to use.
Tasmota does not use any external libraries (Tasmota design rule). Copy the lib you want and add it in a subfolder of `lib` 
You have to change the different [env] accordingly

**Reply by bilogic** [reply]
Haha okay, that explains.. thanks! 

But is `lib_ignore` necessary? I mean, if the header is not referenced, it will not get compiled, i.e. won't add to firmware size right?

**Reply by Jason2866** [reply]
All to save compile time. You are right not included but compiled!
The LDF of platformio cant handle Tasmota completely correctly which parts needs to be compiled...  I invested hours in finding a solution for. No chance. Ivan Kravits (Platformio) knows the limitations. I discussed with him.

**Reply by bilogic** [reply]
Ok understood.

Earlier, I was struggling to figure out why and decided to brute force through by manually copying the `Servo.*` files over. 

Thanks!
